### DISCUSSION TITLE
Execute a bash script to set environment variables when starting gdb

### DISCUSSION THREAD

**Discussion by Gruntfuggly** [microsoft/vscode-cpptools#7947] [General]
I want to use a script to set a bunch of environment variables so I can share my setup when starting my application normally, or when debugging.

Does anybody know if this is possible? I've tried using `preLauchTask` but this seems to just run the task in a separate context so the variables are undefined when it runs the application.

**Comment by sean-mcmanus** [comment]
I know you can set the environment variables in a command environment and then launch VS Code from there and VS Code will inherit those variables, which is the only way cl.exe compilation from VS Code is possible (unless we added some code to update the environment).

You may be able to write an extension that updates the environment and/or create a PR that we could use too -- we're tracking that with https://github.com/microsoft/vscode-cpptools/issues/5271 .

**Reply by Gruntfuggly** [reply]
I already have some of them set in the way you describe, but I also have to set a whole bunch (about 20) in the `env` section of the `launch.json`. These are the ones which I need to share when running under the debugger or normally and it's a pain trying to keep them in sync. It's a shame my project relies to heaving on them 🙁 

I suppose one way to do it would be to actually rewrite the `launch.json` when starting the debugger. That might get a bit messy though.

**Comment by nic-r** [comment]
this is quite a useful requirement
